src/location.js

```diff
@@ -2,6 +2,11 @@ function wrapHistory(keys) {
   return keys.reduce(function(next, key) {
     var fn = history[key]
 
+    // Do not wrap if it's already wrapped
+    if(history[key].toString().indexOf('CustomEvent("pushstate"') !== -1){
+      return function(){};
+    }
```

Comment:
I guess wrapping up a native history object was a bad idea initially...

Reply:
Well, it works well like it is now, but the thing is, you can't use the router 2 times completely separated, because you will wrap the object two times, then you will have 2 copies of every event.